spec/controllers/users_controller_spec.rb

```diff
@@ -95,15 +105,37 @@
     end
   end
 
-  describe 'DELETE users#destroy' do
-    it 'deletes the user' do
+  describe 'users#destroy' do
+    it 'deletes the current_user account' do
       expect(response).to have_http_status(:ok)
       expect { delete :destroy, params: { id: user.id } }.to change(User, :count).by(-1)
     end
 
-    it 'does not delete any user if the user id is invalid' do
+    it 'returns status code forbidden if the user tries to delete another user' do
+      new_user = create(:user)
+      expect { delete :destroy, params: { id: new_user.id } }.not_to change(User, :count)
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'returns status code forbidden if the user id is invalid' do
       expect { delete :destroy, params: { id: 'invalid-id' } }.not_to change(User, :count)
-      expect(response).to have_http_status(:not_found)
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    context 'current_user with ManageUsers permission' do
+      before do
+        user.update!(role: manage_users_role)
+      end
+
+      it 'deletes a user' do
+        new_user = create(:user)
+        expect { delete :destroy, params: { id: new_user.id } }.to change(User, :count).by(-1)
+      end
+
+      it 'returns status code not found if the user id is invalid' do
```

Comment:
I think this will clearify what I'm trying to say,
This is testing a `:not_found` case, an inexistant id isn't invalid.

Reply:
Right. Agreed. I don't find this test particularly useful neither.

Reply:
I changed it to `'returns status code not found if the user does not exists'`